logstash-core/lib/logstash/config/mixin.rb

```diff
@@ -396,7 +394,7 @@ def validate_check_parameter_values(params)
             # Used for converting values in the config to proper objects.
             params[key] = processed_value
           else
-            @logger.error(I18n.t("logstash.runner.configuration.setting_invalid",
+            self.logger.error(I18n.t("logstash.runner.configuration.setting_invalid",
                                  :plugin => @plugin_name, :type => @plugin_type,
                                  :setting => key, :value => value.inspect,
                                  :value_type => config_settings[:validate],
```

Comment:
there is no need to call `self`, The `logger` should be found in the current lookup of the class.